helm: add documentation to deploy with v2 operator (#470)
cockroachdb/helm-charts: operator/README.md

          nodes: 3
          cloudProvider: gcp
          domain: cluster.gke.gcp-us-central1
          namespace: <crdb-namespace>
```

## Upgrade CockroachDB cluster

Modify the config in [`cockroachdb/values.yaml`](/cockroachdb/values.yaml) and upgrade through the below command:

```shell
 $ helm upgrade cockroachdb ./cockroachdb -n cockroach-ns
```

## Scale Up/Down CockroachDB cluster

Update the nodes accordingly under `regions` section and perform the helm upgrade:

```
    regions:
        - code: us-east-1
          nodes: 5
          cloudProvider: k3d
          namespace: default
```

```shell
 $ helm upgrade cockroachdb ./cockroachdb -n cockroach-ns
```

## Rolling Restart of CockroachDB Cluster

Update the timestamp annotation to do a rolling restart of all CockroachDB pods:

```shell
 helm upgrade cockroachdb . --set timestamp=$(date +%s) --reuse-values -n cockroach-ns
```

## Kill a CockroachDB Node

```shell
 kubectl delete pod <pod-name> -n cockroach-ns
```

## Connecting to the CockroachDB cluster

CockroachDB is PostgreSQL wire protocol compatible, so there's a [wide variety of supported clients](https://www.cockroachlabs.com/docs/install-client-drivers.html).

> Once the cluster has been created, you can connect to it through a public Service object created during installation (replace `<public-service>` with the Service name suffixed by `-public`). As an example, we'll use CockroachDB's built-in SQL client as below:

Reviewer: ```suggestion
Once the cluster has been created, you can connect to it through a public Service object created during installation (replace `<public-service>` with the Service name that is suffixed by `-public`). As an example, we'll use CockroachDB's built-in SQL client as below:
```

# CockroachDB Helm Chart

[CockroachDB](https://github.com/cockroachdb/cockroach) - the cloud-native distributed SQL database.

Below is a brief overview of operating the CockroachDB Helm Chart(v2) with Operator.

## Prerequisites

> * Kubernetes 1.8

Reviewer 2: 1.8 is way end of life? Should we update this to be a supported version?

Reviewer: This has been updated to 1.30 now. Thank you for calling this out.

* Helm 3.0 or higher
* Create a namespace to perform the operations against. In this example, we are using `cockroach-ns` namespace.
* If you want to secure your cluster to use TLS certificates for all network communications, [Helm must be installed with RBAC privileges](https://helm.sh/docs/topics/rbac/) or else you will get an "attempt to grant extra privileges" error.

## Installation

### Install Operator

To install the operator chart with the release name `crdb-operator`:

```shell

> $ helm install crdb-operator ./operator -n cockroach-ns

Reviewer 2: Where does one get the operator binary?

```

### Install CockroachDB

- Modify the `regions` config under the `operator` section of [`values.yaml`](/cockroachdb/values.yaml)

```
    regions:
        - code: us-east-1
          nodes: 3

>           cloudProvider: k3d

Reviewer 2: are we assuming people will test this locally with k3d? Would there be value to trying this on other platforms as well?

Author: Its just an example given here, we mentioned relevant gcp example under Multi Region Deployments section.

Reviewer: Let's just put gcp here for consistency.

          namespace: cockroach-ns
```

> - Modify the certs and self-signer config under the `tls` section, as required
> - Modify the other relevant config like `topologySpreadConstraints`, `service.ports`, etc. under the `operator` section, as required
> 
> - If you are bringing your own CA, modify the `certificates.externalCertificates.clientCaConfigMapName` and `certificates.externalCertificates.nodeCaConfigMapName` under the `operator` section with corresponding values

Reviewer: ```suggestion
- Modify the other relevant config like `topologySpreadConstraints`, `service.ports`, etc. under the `operator` section, as required
- Modify the certs and self-signer config under the `tls` section, as required. If you are using a custom CA, update the values for `clientCaConfigMapName` and `nodeCaConfigMapName` under the `operator.certificates.externalCertificates` section
```

Update the nodes accordingly under `regions` section and perform the helm upgrade:

```
    regions:
        - code: us-east-1
          nodes: 5
          cloudProvider: k3d
          namespace: default
```

```shell
 $ helm upgrade cockroachdb ./cockroachdb -n cockroach-ns
```

## Rolling Restart of CockroachDB Cluster

Update the timestamp annotation to do a rolling restart of all CockroachDB pods:

```shell
 helm upgrade cockroachdb . --set timestamp=$(date +%s) --reuse-values -n cockroach-ns
```

## Kill a CockroachDB Node

```shell
 kubectl delete pod <pod-name> -n cockroach-ns
```

## Connecting to the CockroachDB cluster

CockroachDB is PostgreSQL wire protocol compatible, so there's a [wide variety of supported clients](https://www.cockroachlabs.com/docs/install-client-drivers.html).
Once the cluster has been created, you can connect to it through a public Service object created during installation (replace `<public-service>` with the Service name suffixed by `-public`). As an example, we'll use CockroachDB's built-in SQL client as below:

```shell
$ kubectl run cockroach-client --rm -it \
--image=cockroachdb/cockroach \
--restart=Never \

> -- sql --insecure --host <public-service>

Reviewer 2: naive question -- do you have to create the public service endpoint?
